os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -586,7 +564,7 @@ static int i2c_esp32_transmit(struct i2c_dev_s *dev)
 	}
 #endif
 
-	return 0;
+	return ret;
```

Comment:
When `WAIT_ISR` is enabled, return value of this function, `i2c_esp32_transmit` is actually a return value of semaphore.
It looks strange.
But with previous, I could think it is done without any error.

1. I think removing the `ret` variable is better.
```
#if WAIT_ISR
  if (sem_timedwait(xxx) < 0) {
    return -ETIMEDOUT;
  }
  .....
#endif

  return OK;
```
2.  When `WAIT_ISR` is disabled, is this function working well with above?

Reply:
1. Fixed, thanks
2. Yes, we have tested the `WAIT_ISR = 0` branch, it's works well.

Reply:
For 1st, I don't think it is fixed.

1. Without WAIT_ISR enabled, `ret` is used to return 0 always. It looks unnecessary. No need to spend stack.
2. With WAIT_ISR enabled, 'ret' shows a result of semaphore operation only. But it is not enough important for this function.

I think we had not better to use `ret`.
(I agree there is no issue even we use it)

Reply:
Finally you can decide it to keep or to change. Actually it is not important. I will follow your opinion.